src/login/logind-inhibit.c

```diff
@@ -245,11 +230,25 @@ int inhibitor_load(Inhibitor *i) {
                         log_debug_errno(r, "Failed to parse UID of inhibitor: %s", uid);
         }
 
+        pidref_done(&i->pid);
         if (pid) {
-                pidref_done(&i->pid);
                 r = pidref_set_pidstr(&i->pid, pid);
                 if (r < 0)
-                        log_debug_errno(r, "Failed to parse PID of inhibitor: %s", pid);
+                        log_debug_errno(r, "Failed to parse PID of inhibitor, ignoring: %s", pid);
+                else if (pidfdid) {
+                        uint64_t fd_id;
+                        r = safe_atou64(pidfdid, &fd_id);
+                        if (r < 0)
+                                log_warning_errno(r, "Failed to parse leader pidfd ID, ignoring: %s", pidfdid);
+                        else {
+                                (void) pidref_acquire_pidfd_id(&i->pid);
```

Comment:
So, hmm, I'm pretty sure we should also submit pidfds to fdstore, not simply storing pidfdid here. The thing is that I'd like to introduce inhibitors bound to certain pidfd and not explicitly installed fifo later, which can only work reliably if the pidfd is serialized, not pidfdid. Hence if we will be doing the pidfd dance already for some type of inhibitors, we might as well just use it everywhere.